server/src/main/java/org/apache/iotdb/db/query/executor/LastQueryExecutor.java

```diff
@@ -145,13 +145,17 @@ public static TimeValuePair calculateLastPairForOneSeriesLocally(
         TimeValuePair timeValuePair = IoTDB.metaManager.getLastCache(seriesPath);
         if (timeValuePair != null && satisfyFilter(filter, timeValuePair)) {
           return timeValuePair;
+        } else if (timeValuePair != null) {
+          return null;
         }
       }
 
       if (node != null && node.getCachedLast() != null) {
         TimeValuePair timeValuePair =  node.getCachedLast();
         if (timeValuePair != null && satisfyFilter(filter, timeValuePair)) {
           return timeValuePair;
+        } else if (timeValuePair != null) {
+          return null;
```

Comment:
If the cached last time-value pair doesn't satify the filter, shouldn't we keep to call the calculateLastPairByScanningTsFiles() function?